beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -139,11 +140,92 @@ impl<T: EthSpec> PeerInfo<T> {
         self.enr.as_ref()
     }
 
+    /// An iterator over all the subnets this peer is subscribed to.
+    pub fn subnets(&self) -> impl Iterator<Item = &Subnet> {
+        self.subnets.iter()
+    }
+
+    /// Returns the number of long lived subnets a peer is subscribed to.
+    // NOTE: This currently excludes sync committee subnets
+    pub fn long_lived_subnet_count(&self) -> usize {
+        if let Some(meta_data) = self.meta_data.as_ref() {
+            return meta_data.attnets().num_set_bits();
+        } else if let Some(enr) = self.enr.as_ref() {
+            if let Ok(attnets) = enr.attestation_bitfield::<T>() {
+                return attnets.num_set_bits();
+            }
+        }
+        0
+    }
+
+    /// Returns an iterator over the long-lived subnets if it has any.
+    pub fn long_lived_subnets(&self) -> Vec<Subnet> {
+        let mut long_lived_subnets = Vec::new();
+        // Check the meta_data
+        if let Some(meta_data) = self.meta_data.as_ref() {
+            for subnet in 0..=meta_data.attnets().highest_set_bit().unwrap_or(0) {
+                if meta_data.attnets().get(subnet).expect("within bounds") {
```

Comment:
```suggestion
                if meta_data.attnets().get(subnet).unwrap_or(false) {
```

May as well avoid the `expect` when we have a safe alternative. I thought for a second that this could panic when the ENR had no bits set, but because it's a fixed-length bit vector it's always safe to index it at 0.